actors/verifreg/src/events.rs

```diff
@@ -0,0 +1,105 @@
+use crate::{Allocation, AllocationID, Claim, ClaimID, DataCap};
+use fil_actors_runtime::runtime::Runtime;
+use fil_actors_runtime::{ActorError, EventBuilder};
+use fvm_shared::ActorID;
+
+// A namespace for helpers that build and emit verified registry events.
+// REVIEW: would this be better as a simple module with pub fns, called as emit::verifier_balance()?
+pub struct Emit {}
+
+impl Emit {
+    /// Indicates a new value for a verifier's datacap balance.
+    /// Note that receiving this event does not necessarily mean the balance has changed.
+    /// The value is in datacap whole units (not TokenAmount).
+    pub fn verifier_balance(
+        rt: &impl Runtime,
+        verifier: ActorID,
+        new_balance: &DataCap,
+    ) -> Result<(), ActorError> {
+        rt.emit_event(
+            &EventBuilder::new()
+                .label("verifier-balance")
+                .value_indexed("verifier", &verifier)?
+                .value("balance", new_balance)?
+                .build(),
+        )
```

Comment:
I'd prefer to be able to write:

```suggestion
        rt.build_event()
                .label("verifier-balance") // or maybe _require_ a label and take it as a parameter to build_event?
                .field_indexed("verifier", &verifier)
                .field("balance", new_balance)
                .emit()?;
```

If we also make the event builder an associated type on the runtime (so it can be different in test code versus when run inside the FVM), this could be _really_ efficient and avoid a bunch of allocations. Once we land https://github.com/filecoin-project/ref-fvm/pull/1807, we'll be able to build/emit events with very few copies/allocations.

Reply:
(note: I'm calling them "fields" because that's the term used by most structured loggers)

Reply:
NOTE: An associated type wouldn't preclude traits like `WithAllocation`. You'd just have to implement them as follows:

```rust
impl<E: EventBuilder> WithAllocation for E {
    ....
}

Reply:
I am not enthusiastic at all about adding anything to the runtime. The code in runtime that's not just directly calling through to syscalls is just library code and shouldn't have anything to do with the syscall interface, or have a privileged name. Everything inside it is basically untestable here and I think it should migrate towards being the thinnest possible layer.

We can of course rework patterns for emitting events if we get a new syscall for it.

Reply:
I have change the naming to use "field".